hs_core/tasks.py

```diff
@@ -166,6 +167,15 @@ def create_bag_by_irods(resource_id, istorage=None):
     from hs_core.hydroshare.utils import get_resource_by_shortkey
 
     res = get_resource_by_shortkey(resource_id)
+    try:
+        # we will create xml metadata files every time prior to creating the bag
+        # the assumption here is that bag is being created due to the resource
+        # modified (AVU)
+        create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
+    except Exception as ex:
+        logger.error('Failed to create bag files. Error:{}'.format(ex.message))
+        return False
+

```

Comment:
I think it is better to add the additional check to only update it when needed. This is why the metadata_dirty flag is added in the first place. If there is an issue/bug with metadata_dirty flag, we need to fix it rather than doing unneeded operations to make sure just in case. If the metadata_dirty flag is correct, then xml file content should be correct/up-to-date with additional flag. With async issue that at the exact time when one user is updating the metadata, another user is downloading the bag, if the xml file does not include that user's update, I think that is expected because of timing.